niri-config/src/lib.rs

```diff
@@ -183,6 +185,8 @@ pub struct Mouse {
     pub accel_speed: f64,
     #[knuffel(child, unwrap(argument, str))]
     pub accel_profile: Option<AccelProfile>,
+    #[knuffel(child, unwrap(argument, str))]
+    pub scroll_method: Option<ScrollMethod>,
```

Comment:
So this does make sense on a mouse device? Rather than just touchpad and trackpoint.

Reply:
> So this does make sense on a mouse device?

Yes. The supported methods vary depending on devices, though. For example, my USB keyboard has pointing stick, but the device is recognized as mouse (on both Niri and Sway.)
```
Device:           PFU Limited HHKB-Studio
Kernel:           /dev/input/event5
Group:            6
Seat:             seat0, default
Capabilities:     pointer 
Tap-to-click:     n/a
Tap-and-drag:     n/a
Tap drag lock:    n/a
Left-handed:      disabled
Nat.scrolling:    disabled
Middle emulation: disabled
Calibration:      n/a
Scroll methods:   button
Click methods:    none
Disable-w-typing: n/a
Disable-w-trackpointing: n/a
Accel profiles:   flat *adaptive custom
Rotation:         0.0
```

Perhaps, the device type can be fixed by udev hwdb, but I guess physical mouse also supports `on-button-down` method.